Start busybox crond in the entrypoint.sh

brunorro opened the pull request:
I found it the easiest way to update stats in kubernetes: the cron files should 
be bind-mounted on /etc/periodic/{15min,daily,hourly,monthly,weekly}/script.
The crond process running on the background should use <2MB resident 
memory (on x86_64)

Commit:
Allow the execution of an autorun.sh in the entrypoint

If a file /usr/local/bin/autorun.sh is bind-mounted will be executed during the start
of the container. That gives more elasticity than only starting a cron daemon.

Commit:
Start busybox crond in the entrypoint.sh

The crond provided with busybox is started in the entrypoint. That can
be useful when the container is deployed in kubernetes: the cron files
can be bind-mounted on /etc/periodic/{15min,daily,hourly,monthly,weekly}

pabra commented:
Thanks for contributing, but I do not really like the idea of having `crond` running disconnected - without init/supervisor/foreman - in the background. What if `crond` dies? I know that this might not be very likely.
Also I do not like the idea of having more than one service running in one container.
After all I (personally) do not like having `crond` running inside the container - I prefer to trigger jobs from outside.

If you do not care about `crond` in the background without supervisor, you can still bind-mount your /etc/periodic scripts and start `crond` after the container started like 
```console
docker exec awstats /usr/sbin/crond -L /var/log/crond.log -l 8
```

I would accept your PR if `crond` (or whatever else) would be optional. Instead of starting `crond` directly in the `entrypoint.sh`, you could test if there is an executable file e.g. at `/usr/local/bin/autorun.sh` and if so, execute it. That way you can bind-mount a schell script to `/usr/local/bin/autorun.sh` and in there start whatever you like.

brunorro commented:
Hello! Thanks a lot for your reply

Your approach (one container=one service) is the right one and the one I like too, I know this pull request was made a bit of a sudden. I'll check your recommendation about the autorun.sh, it sounds much better (and customizable) than force the crond starting

Thank you again

Commit:
Allow the execution of an autorun.sh in the entrypoint

If a file /usr/local/bin/autorun.sh is bind-mounted will be executed during the start
of the container. That gives more elasticity than only starting a cron daemon.

Commit:
Allow the execution of an autorun.sh in the entrypoint

If a file /usr/local/bin/autorun.sh is bind-mounted will be executed during the start
of the container. That gives more elasticity than only starting a cron daemon.

Commit:
Allow the execution of an autorun.sh in the entrypoint

If a file /usr/local/bin/autorun.sh is bind-mounted will be executed during the start
of the container. That gives more elasticity than only starting a cron daemon.

Commit:
Allow the execution of an autorun.sh in the entrypoint

If a file /usr/local/bin/autorun.sh is bind-mounted will be executed during the start
of the container. That gives more elasticity than only starting a cron daemon.
